### PR TITLE
feat: Import favorites from legacy

### PR DESCRIPTION
**Pull request by ryanzhoudev**
Adds + button to guide list page. Imports favorites as exported in https://github.com/Wynntils/Wynntils/pull/681. Review and merge together.

### CODE REVIEW

File: common/src/main/java/com/wynntils/screens/base/widgets/ImportButton.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.screens.base.TooltipProvider;
+import com.wynntils.utils.render.RenderUtils;
+import com.wynntils.utils.render.Texture;
+import java.util.List;
+import net.minecraft.ChatFormatting;
+import net.minecraft.network.chat.Component;
+
+public class ImportButton extends WynntilsButton implements TooltipProvider {
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/screens/base/widgets/ImportButton.java]: > implements TooltipProvider

haven't seen this one before, I usually manually rendered them.. Do they work as they are supposed to?

File: common/src/main/java/com/wynntils/screens/guides/WynntilsGuidesListScreen.java [hunk 2]
```diff
@@ -89,6 +122,24 @@ public void doRender(PoseStack poseStack, int mouseX, int mouseY, float partialT
...
+        if (tooltipLines.isEmpty()) return;
+
+        RenderUtils.drawTooltipAt(
+                poseStack,
+                mouseX,
+                mouseY,
+                100,
+                tooltipLines,
+                FontRenderer.getInstance().getFont(),
+                true);
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/screens/guides/WynntilsGuidesListScreen.java:156]: Oh, we still render it manually. whatever then

**Review by kristofbolyai — COMMENTED** [review]
If it is not hard, can you implement exporting to Artemis as well? I assume it wouldn't be hard, but would be useful.

**Comment by kristofbolyai** [comment]
> If it is not hard, can you implement exporting to Artemis as well? I assume it wouldn't be hard, but would be useful.

If not, let me know and I will review/approve.

**Comment by kristofbolyai** [comment]
> > If it is not hard, can you implement exporting to Artemis as well? I assume it wouldn't be hard, but would be useful.
> 
> If not, let me know and I will review/approve.

I just saw that we only had a command for this. So we don't have a GUI design.. I guess that can wait.

**Comment by ryanzhoudev** [comment]
> > > If it is not hard, can you implement exporting to Artemis as well? I assume it wouldn't be hard, but would be useful.
> > 
> > If not, let me know and I will review/approve.
> 
> I just saw that we only had a command for this. So we don't have a GUI design.. I guess that can wait.

Yeah, I didn't think it was worth it to work on legacy GUI. I can add a button under the import button if you want.

**Comment by kristofbolyai** [comment]
Sure, if you want to

File: common/src/main/java/com/wynntils/screens/guides/WynntilsGuidesListScreen.java [hunk 3]
```diff
@@ -68,6 +90,35 @@ protected void doInit() {
...
+        });
+        McUtils.sendMessageToClient(
+                Component.translatable("screens.wynntils.wynntilsGuides.importedFavorites", names.size())
+                        .withStyle(ChatFormatting.GREEN));
+    }
+
+    private void exportFavorites() {
+        McUtils.mc()
+                .keyboardHandler
+                .setClipboard("wynntilsFavorites" + String.join(",", Models.Favorites.getFavoriteItems()));
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/screens/guides/WynntilsGuidesListScreen.java]: Is this right? doesn't this produce "wynntilsFavoritesa,b" for [a, b]? (Does this work for legacy as well)

**Review by kristofbolyai — COMMENTED** [review]
Looks good otherwise

File: common/src/main/java/com/wynntils/screens/guides/WynntilsGuidesListScreen.java (hunk 3 above)
Comment (ryanzhoudev) [reply common/src/main/java/com/wynntils/screens/guides/WynntilsGuidesListScreen.java]: Oh yeah my bad, forgot the comma. It's fine on legacy.

File: common/src/main/java/com/wynntils/screens/guides/WynntilsGuidesListScreen.java [hunk 4]
```diff
@@ -64,6 +87,35 @@ protected void doInit() {
                 this));
     }
 
+    private void importFavorites() {
+        String clipboard = McUtils.mc().keyboardHandler.getClipboard();
+
+        if (clipboard == null || !clipboard.startsWith("wynntilsFavorites,")) {
+            McUtils.sendMessageToClient(Component.translatable("screens.wynntils.wynntilsGuides.invalidClipboard")
```
Comment (magicus) [inline common/src/main/java/com/wynntils/screens/guides/WynntilsGuidesListScreen.java]: You might want to use the new sendErrorToClient instead.

**Review by magicus — APPROVED** [review]

**Pull request merged**